diff --git a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionFailedEvent.kt b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionFailedEvent.kt
index 192fc62..919ba76 100644
--- a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionFailedEvent.kt
+++ b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionFailedEvent.kt
@@ -5,10 +5,10 @@ import com.facebook.react.uimanager.events.Event
 import com.facebook.react.uimanager.events.RCTEventEmitter
 import video.api.reactnative.livestream.ViewProps
 
-class OnConnectionFailedEvent(private val viewTag: Int, private val reason: String?) :
-  Event<OnConnectionFailedEvent>(viewTag) {
+class OnConnectionFailedEvent(tag: Int, private val code: String?) :
+  Event<OnConnectionFailedEvent>(tag) {
   private val params = Arguments.createMap().apply {
-    putString("code", reason)
+    putString("code", code)
   }
 
   override fun getEventName() = ViewProps.Events.CONNECTION_FAILED.eventName
diff --git a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionSuccessEvent.kt b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionSuccessEvent.kt
index d99ff18..68018ef 100644
--- a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionSuccessEvent.kt
+++ b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnConnectionSuccessEvent.kt
@@ -1,13 +1,15 @@
 package video.api.reactnative.livestream.events
 
+import com.facebook.react.bridge.Arguments
 import com.facebook.react.uimanager.events.Event
 import com.facebook.react.uimanager.events.RCTEventEmitter
 import video.api.reactnative.livestream.ViewProps
 
-class OnConnectionSuccessEvent(private val viewTag: Int): Event<OnConnectionSuccessEvent>(viewTag) {
+class OnConnectionSuccessEvent(tag: Int) :
+  Event<OnConnectionSuccessEvent>(tag) {
   override fun getEventName() = ViewProps.Events.CONNECTION_SUCCESS.eventName
 
   override fun dispatch(rctEventEmitter: RCTEventEmitter) {
-    rctEventEmitter.receiveEvent(viewTag, eventName, null)
+    rctEventEmitter.receiveEvent(viewTag, eventName, Arguments.createMap())
   }
 }
diff --git a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnDisconnectEvent.kt b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnDisconnectEvent.kt
index a3da124..b1cec80 100644
--- a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnDisconnectEvent.kt
+++ b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnDisconnectEvent.kt
@@ -1,14 +1,15 @@
 package video.api.reactnative.livestream.events
 
+import com.facebook.react.bridge.Arguments
 import com.facebook.react.uimanager.events.Event
 import com.facebook.react.uimanager.events.RCTEventEmitter
 import video.api.reactnative.livestream.ViewProps
 
-class OnDisconnectEvent(private val viewTag: Int) :
-  Event<OnDisconnectEvent>(viewTag) {
-  override fun getEventName() = ViewProps.Events.DISCONNECTED.eventName
+class OnDisconnectEvent(tag: Int) :
+  Event<OnDisconnectEvent>(tag) {
+  override fun getEventName() = "onDisconnect"
 
   override fun dispatch(rctEventEmitter: RCTEventEmitter) {
-    rctEventEmitter.receiveEvent(viewTag, eventName, null)
+    rctEventEmitter.receiveEvent(viewTag, eventName, Arguments.createMap())
   }
 }
diff --git a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnPermissionsDeniedEvent.kt b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnPermissionsDeniedEvent.kt
index 7873b35..fa8418b 100644
--- a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnPermissionsDeniedEvent.kt
+++ b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnPermissionsDeniedEvent.kt
@@ -5,15 +5,16 @@ import com.facebook.react.uimanager.events.Event
 import com.facebook.react.uimanager.events.RCTEventEmitter
 import video.api.reactnative.livestream.ViewProps
 
-class OnPermissionsDeniedEvent(private val viewTag: Int, private val permissions: List<String>) :
-  Event<OnPermissionsDeniedEvent>(viewTag) {
-  private val params = Arguments.createMap().apply {
-    putArray("permissions", Arguments.fromList(permissions))
-  }
-
+class OnPermissionsDeniedEvent(tag: Int, private val permissions: List<String>) :
+  Event<OnPermissionsDeniedEvent>(tag) {
   override fun getEventName() = ViewProps.Events.PERMISSIONS_DENIED.eventName
 
   override fun dispatch(rctEventEmitter: RCTEventEmitter) {
+    val params = Arguments.createMap().apply {
+      val array = Arguments.createArray()
+      permissions.forEach { array.pushString(it) }
+      putArray("permissions", array)
+    }
     rctEventEmitter.receiveEvent(viewTag, eventName, params)
   }
 }
diff --git a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnStartStreamingEvent.kt b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnStartStreamingEvent.kt
index 6cf2bbd..eb0eda6 100644
--- a/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnStartStreamingEvent.kt
+++ b/node_modules/@api.video/react-native-livestream/android/src/main/java/video/api/reactnative/livestream/events/OnStartStreamingEvent.kt
@@ -6,21 +6,19 @@ import com.facebook.react.uimanager.events.RCTEventEmitter
 import video.api.reactnative.livestream.ViewProps
 
 class OnStartStreamingEvent(
-  private val viewTag: Int,
-  private val requestId: Int,
+  viewId: Int,
+  private val requestId: Any?,
   private val result: Boolean,
-  private val error: String? = null
-) :
-  Event<OnStartStreamingEvent>(viewTag) {
-  private val params = Arguments.createMap().apply {
-    putInt("requestId", requestId)
-    putBoolean("result", result)
-    error?.let { putString("error", it) }
-  }
-
+  private val error: Any?
+) : Event<OnStartStreamingEvent>(viewId) {
   override fun getEventName() = ViewProps.Events.START_STREAMING.eventName
 
   override fun dispatch(rctEventEmitter: RCTEventEmitter) {
+    val params = Arguments.createMap().apply {
+      putString("requestId", requestId.toString())
+      putBoolean("result", result)
+      putString("error", error?.toString())
+    }
     rctEventEmitter.receiveEvent(viewTag, eventName, params)
   }
 }
